### ISSUE TITLE
Widgets inside editor should come last in their order of visibility when tabbing

### ISSUE THREAD

**Issue by bpasero** [Microsoft/vscode#2292]
When I use Tab to tab into the editor and have widgets open (e.g. find widget, view zone), focus jumps first between these widgets and then into the editor. I would think that focus should first move into the editor and then to the widgets in their order.

**Comment by alexandrudima** [comment]
@bpasero Since focus follows DOM order, I cannot do anything on my side as a general fix to make focus be reveal order dependent. Maybe you can give a specific use-case and sequence of steps and then we can see what we can do.

**Comment by bpasero** [comment]
@alexandrudima can we not use the tabindex property with a non-zero positive value to control the tab navigation within the editor? I filed this bug originally because I was expecting to first tab into the editor (from the outside) before tabbing e.g. into the find widget.

**Comment by bpasero** [comment]
I am also not sure why the find widget would appear before the editor, isnt the editor the container of the find widget in the end?

**Comment by alexandrudima** [comment]
A tabIndex solution would not work because tabIndex is, similar to id, a global property.

E.g. if an editor is on slot 1 and it uses tabIndex from 1 to 10 for different things, then the editor on slot 2 would need to use tabIndex 11 to 20, etc. This would also need to be aligned with our other usages of tabIndex = 0, otherwise you would loop for a long time amongst all the things with tabIndex = 0.

So you'd expect when tabbing from the explorer to drop in the editor's text?

**Comment by bpasero** [comment]
Good point regarding multiple editors. When you tab out of the explorer we first tab into the editor tool bar. After that you end up tabbing into the find widget if that is open and then to the editor.

As you say, since the textarea comes after the widgets, it makes sense. Can we not move the textarea before the editor widgets container?
